raster/src/main/scala/geotrellis/raster/summary/polygonal/PolygonalSummaryResult.scala

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2019 Azavea
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package geotrellis.raster.summary.polygonal
+
+import cats.{Monad, Semigroup}
+import cats.syntax.semigroup._
+
+import scala.annotation.tailrec
+import scala.reflect.ClassTag
+
+/**
+  * A Result ADT returned by [[PolygonalSummary]] operations
+  *
+  * There are two result types, [[NoIntersection]] and [[Summary]].
+  * This ADT will always return Summary if the Raster and Polygon intersect.
+  * Otherwise, NoIntersection will be returned.
+  *
+  * Provides helpers to convert to Option and Either if you don't need to match
+  * on result type.
+  */
+sealed trait PolygonalSummaryResult[+A] {
+  def toOption: Option[A]
+
+  def toEither: Either[Any, A]
```

Comment:
`Either[Any, A]` is extremely weird. The point of `Either` is to preserve type information between two outcomes that don't have a useful upper bound. Placing `Any` on the left side basically erases any type information. Just looking at it maybe the better thing is `Either[NonIntersecting, A]` .. that `NonIntersecting` have to become `GeometryError` later if we try to capture JTS errors that result when trying to intersect invalid geometries.

If there isn't strong need for `Either` in this case we could also just choose to dump this method until there is.

Reply:
Whoops. Missed this one in my cleanup pass. Updated to NoIntersection.